feat: message propagation metrics

deffrian opened the pull request:
Ref: A-413, A-412

Span to track attestation propagation on `AttestToProposal` function
Peer connection time metric


Also changed metrics from previous pr:
1. track deleted transactions in `cleanupDeletedMinedTxs`
3. Record p2p requested transaction even if we requested zero

mralj reviewed (changes requested):
@deffrian, can you please look into the [flaky test](http://ci.aztec-labs.com/f3e561abfd5fdc4d)? I believe this PR introduced it.

deffrian committed:
feat: message propagation metrics

Ref: A-413, A-412

Span to track attestation propagation on `AttestToProposal` function
Peer connection time metric

Also changed metrics from previous pr:
1. track deleted transactions in `cleanupDeletedMinedTxs`
3. Record p2p requested transaction even if we requested zero

Co-authored-by: Nikita Meshcheriakov <root@nikitam.io>
